src/arrayMethodSort.js

```diff
@@ -4,8 +4,31 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function (compareFunction) {
     // write code here
+    const compare =
+      compareFunction ||
+      function (a, b) {
+        if (String(a) > String(b)) {
+          return 1;
+        }
+
+        if (String(a) < String(b)) {
+          return -1;
+        }
+
+        return 0;
+      };
+
+    for (let i = 0; i < this.length; i++) {
```

Comment:
The outer loop also doesn't need to iterate over the entire array. Once the largest elements have been 'bubbled' to the end of the array, they are in their final positions and don't need to be considered in further iterations. You can stop the outer loop when no more swaps are made. To do this, you can add a variable `swapped` before the loop that tracks whether any elements have been swapped during each pass. If a pass completes without any swaps, you can break out of the loop early.